Source/Core/Core/NetworkCaptureLogger.cpp

```diff
@@ -0,0 +1,390 @@
+  ip_header[8] = 0x40;
+  // Write protocol
+  ip_header[9] = protocol;
+
+  // Get socket IP address
+  struct sockaddr_in addr;
+  socklen_t addr_len = sizeof(struct sockaddr_in);
+  getpeername(socket, reinterpret_cast<struct sockaddr*>(&addr), &addr_len);
+
+  if (is_sender)
+  {
+    // Write Source IP address: 127.0.0.1
+    ip_header[12] = 0x7F;
+    ip_header[15] = 0x01;
+    // Write Destination IP address
+    std::memcpy(&ip_header[16], &addr.sin_addr, 4);
+  }
+  else
+  {
+    // Write Source IP address
+    std::memcpy(&ip_header[12], &addr.sin_addr, 4);
+    // Write Destination IP address: 127.0.0.1
+    ip_header[16] = 0x7F;
+    ip_header[19] = 0x01;
+  }
+
+  // Compute checksum
+  std::size_t index = 0;
+  u32 checksum = 0;
+  u16 word = 0;
+  for (u8 b : ip_header)
+  {
+    if (index % 2 == 0)
+      word = b;
+    else
+      checksum += (word << 8) + b;
+    index += 1;
+  }
+  checksum = (checksum + (checksum >> 16) & 0xFFFF) ^ 0xFFFF;
+
+  // Write checksum
+  ip_header[10] = checksum >> 8 & 0xFF;
+  ip_header[11] = checksum & 0xFF;
+
+  return ip_header;
+}
+
+void PCAPSSLCaptureLogger::LogTCP(bool is_sender, const void* data, std::size_t length, s32 socket)
+{
+  constexpr std::size_t TCP_HEADER_SIZE = 20;
+  std::vector<u8> tcp_header(TCP_HEADER_SIZE);
+
+  // Get socket port
+  struct sockaddr_in sock_addr;
+  struct sockaddr_in peer_addr;
+  socklen_t addr_len = sizeof(struct sockaddr_in);
+  getsockname(socket, reinterpret_cast<struct sockaddr*>(&sock_addr), &addr_len);
+  getpeername(socket, reinterpret_cast<struct sockaddr*>(&peer_addr), &addr_len);
+
+  // Write port
+  if (is_sender)
+  {
+    std::memcpy(&tcp_header[0], &sock_addr.sin_port, 2);
+    std::memcpy(&tcp_header[2], &peer_addr.sin_port, 2);
+
+    // Write sequence number
+    static u32 sequence_number = 0;
+    tcp_header[4] = sequence_number >> 24 & 0xFF;
+    tcp_header[5] = sequence_number >> 16 & 0xFF;
+    tcp_header[6] = sequence_number >> 8 & 0xFF;
+    tcp_header[7] = sequence_number & 0xFF;
+    sequence_number += static_cast<u32>(length);
+  }
+  else
+  {
+    std::memcpy(&tcp_header[0], &peer_addr.sin_port, 2);
+    std::memcpy(&tcp_header[2], &sock_addr.sin_port, 2);
+
+    // Write sequence number
+    static u32 sequence_number = 0;
+    tcp_header[4] = sequence_number >> 24 & 0xFF;
+    tcp_header[5] = sequence_number >> 16 & 0xFF;
+    tcp_header[6] = sequence_number >> 8 & 0xFF;
+    tcp_header[7] = sequence_number & 0xFF;
+    sequence_number += static_cast<u32>(length);
+  }
+
+  // Write data offset
+  tcp_header[12] = 0x50;
+
+  // Write flags
+  // TODO
+
+  // Write window size
+  tcp_header[14] = 0xFF;
+  tcp_header[15] = 0xFF;
+
+  // Add the packet
+  std::vector<u8> ethernet_header = GetEthernetHeader(is_sender);
+  std::vector<u8> ip_header = GetIPHeader(is_sender, TCP_HEADER_SIZE + length, 0x06, socket);
+  std::vector<u8> packet;
+  packet.insert(packet.end(), std::make_move_iterator(ethernet_header.begin()),
```

Comment:
`std::make_move_iterator` isn't needed here. There's no internal data within a `u8` that can be moved, it's just a primitive. There's no real difference between a move or a copy in this instance (either way you pass around the same amount of data).

`std::make_move_iterator` is beneficial when a type contains other data that can be moved out of it (for example, a `std::vector<std::string>` would be a good use-case, since the internal character buffer of each `std::string` instance be moved instead of being copied, saving on allocations).